lib/src/matrix/base_matrix.dart

```diff
@@ -156,6 +161,58 @@ abstract class BaseMatrix with
     return Matrix.fromRows(checked, dtype: dtype);
   }
 
+  @override
+  Vector mean([Axis axis = Axis.columns]) {
+    if (columnsNum == 0 || rowsNum == 0) {
+      return Vector.empty(dtype: dtype);
+    }
+
+    switch (axis) {
+      case Axis.columns:
+        return _meansCache[axis] ??= _mean(rows, rowsNum);
+
+      case Axis.rows:
+        return _meansCache[axis] ??= _mean(columns, columnsNum);
+
+      default:
+        throw UnimplementedError('Means calculation for axis $axis is not '
+            'supported yet');
+    }
+  }
+
+  Vector _mean(Iterable<Vector> vectors, int vectorsNum) =>
```

Comment:
use the Vector's `mean` method instead of re-implementing calculation of mean values here